js/ui/grid_core/ui.grid_core.rows.js

```diff
@@ -320,72 +320,83 @@ export const rowsModule = {
-                        const changeType = change && change.changeType;
-                        const executors = [];
-                        const highlightChanges = this.option('highlightChanges');
-                        const rowInsertedClass = this.addWidgetPrefix(ROW_INSERTED_ANIMATION_CLASS);
-
-                        switch(changeType) {
-                            case 'update':
-                                each(change.rowIndices, (index, rowIndex) => {
-                                    const $newRowElement = this._getRowElements(newTableElement).eq(index);
-                                    const changeType = change.changeTypes && change.changeTypes[index];
-                                    const item = change.items && change.items[index];
-
-                                    executors.push(() => {
-                                        const $rowsElement = this._getRowElements();
-                                        const $rowElement = $rowsElement.eq(rowIndex);
-
-                                        switch(changeType) {
-                                            case 'update':
-                                                if(item) {
-                                                    const columnIndices = change.columnIndices && change.columnIndices[index];
-                                                    if(isDefined(item.visible) && item.visible !== $rowElement.is(':visible')) {
-                                                        $rowElement.toggle(item.visible);
-                                                    } else if(columnIndices) {
-                                                        this._updateCells($rowElement, $newRowElement, columnIndices);
+                        const contentChanges = this._contentChanges;
+
+                        this._contentChanges = [];
+
+                        contentChanges.forEach(({ newTableElement, change, isFixedTableRendering }) => {
+                            const tableElement = this.getTableElement(isFixedTableRendering);
+                            const contentElement = this._findContentElement(isFixedTableRendering);
+                            const changeType = change?.changeType;
+                            const executors = [];
+                            const highlightChanges = this.option('highlightChanges');
+                            const rowInsertedClass = this.addWidgetPrefix(ROW_INSERTED_ANIMATION_CLASS);
+
+                            switch(changeType) {
+                                case 'update':
+                                    each(change.rowIndices, (index, rowIndex) => {
+                                        const $newRowElement = this._getRowElements(newTableElement).eq(index);
+                                        const dataChangeType = change.changeTypes?.[index];
+                                        const item = change.items && change.items[index];
```

Comment:
Minor comment.

Let's use ```?.``` here too:
```suggestion
                                        const item = change.items?.[index];
```